esmvaltool/diag_scripts/hydrology/marrmot.py

```diff
@@ -0,0 +1,245 @@
+                                 units='degC')
+    exponent = iris.analysis.maths.exp(emp_a * temp / (emp_b + temp))
+    # return emp_a * emp_b * e0 * exponent / (emp_b + temp)**2
+    # iris.exceptions.NotYetImplementedError: coord * coord (emp_b * e0)
+    # workaround:
+    tmp1 = emp_a * emp_b
+    tmp2 = e0_const * exponent / (emp_b + temp)**2
+    return tmp1 * tmp2
+
+
+def get_constants(psl):
+    """
+    The Definition of rv and rd constants is provided in
+    Wallace and Hobbs (2006), 2.6 equation 3.14.
+    The Definition of lambda and cp is provided in Wallace and Hobbs 2006.
+    The Definition of beta and cs is provided in De Bruin (2016), section 4a.
+    """
+    # Definition of constants
+    rv_const = iris.coords.AuxCoord(461.51,
+                                    long_name='Gas constant water vapour',
+                                    units='J K-1 kg-1')
+
+    rd_const = iris.coords.AuxCoord(287.0,
+                                    long_name='Gas constant dry air',
+                                    units='J K-1 kg-1')
+
+    # Latent heat of vaporization in J kg-1 (or J m-2 day-1)
+    lambda_ = iris.coords.AuxCoord(2.5e6,
+                                   long_name='Latent heat of vaporization',
+                                   units='J kg-1')
+
+    # Specific heat of dry air constant pressure
+    cp_const = iris.coords.AuxCoord(1004,
+                                    long_name='Specific heat of dry air',
+                                    units='J K-1 kg-1')
+
+    beta = iris.coords.AuxCoord(20,
+                                long_name='Correction Constant',
+                                units='W m-2')
+
+    cs_const = iris.coords.AuxCoord(110,
+                                    long_name='Empirical constant',
+                                    units='W m-2')
+
+    # gamma = rv/rd * cp*msl/lambda_
+    # iris.exceptions.NotYetImplementedError: coord / coord
+    gamma = rv_const.points[0] / rd_const.points[0] * cp_const * psl / lambda_
+    return gamma, cs_const, beta, lambda_
+
+
+def debruin_pet(var_dict):
+    """ Determine De Bruin (2016) reference evaporation
+    Implement equation 6 from De Bruin (10.1175/JHM-D-15-0006.1)
+    """
+    # Unit checks:
+    psl = preproc.convert_units(var_dict['psl'], 'hPa')
+    tas = preproc.convert_units(var_dict['tas'], 'degC')
+
+    # Variable derivation
+    delta_svp = tetens_derivative(tas)
+    gamma, cs_const, beta, lambda_ = get_constants(psl)
+
+    # the definition of the radiation components according to the paper:
+    kdown = var_dict['rsds']
+    kdown_ext = var_dict['rsdt']
+    # Equation 6
+    rad_term = (1-0.23)*kdown - cs_const*kdown/kdown_ext
+    # the unit is W m-2
+    ref_evap = delta_svp / (delta_svp + gamma) * rad_term + beta
+
+    # 1 W m-2 = 86400 J m-2 day-1
+    pet = ref_evap * 86400 / lambda_
+    pet.var_name = 'potential_evapotranspiration'
+    pet.units = 'kg m-2 day-1'  # equivalent to mm/day
```

Comment:
It would be safer to use `pet.convert_units('kg m-2 day-1')` instead of manually programming the number of seconds in a day and then assigning the units, because that way you get an error message if something is wrong with the input data (e.g. the units are not compatible).

Reply:
Thanks @bouweandela. The term (* 86400) is a part of the equations. Without this term, the division by lamdda_ is not valid.